### PR TITLE
[Reproducible Builds] Diff in generated/metadata between the builds

### LINKED ISSUE #33928: [Issue] [Reproducible Builds] Diff in generated/metadata between the builds
This issue is automatically created based on existing pull request: magento/magento2#33903: [Reproducible Builds] Diff in generated/metadata between the builds

---------
Sort all configs in generated/metadata before writing to file

### Description (*)
This is a follow-up on https://github.com/magento/magento2/issues/23324 for the 2.4-develop branch.

It's really important to have [Reproducible Builds](https://reproducible-builds.org/). It will help to check diff between the builds, so you'll be sure that you deploy only needed changes to production.

The same steps to reproduce, Magento 2.3.7-p1 or 2.4-develop branch.

The diff between the builds is following:
```
Files build-1/generated/metadata/app_action_list.php and build-2/generated/metadata/app_action_list.php differ
Files build-1/generated/metadata/interception.php and build-2/generated/metadata/interception.php differ
```

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by ihor-sviziev**
Sort all configs in generated/metadata before writing to file

### Description (*)
This is a follow-up on https://github.com/magento/magento2/issues/23324 for the 2.4-develop branch.

It's really important to have [Reproducible Builds](https://reproducible-builds.org/). It will help to check diff between the builds, so you'll be sure that you deploy only needed changes to production.

The same steps to reproduce, Magento 2.3.7-p1 or 2.4-develop branch.

The diff between the builds is following:
```
Files build-1/generated/metadata/app_action_list.php and build-2/generated/metadata/app_action_list.php differ
Files build-1/generated/metadata/interception.php and build-2/generated/metadata/interception.php differ
```

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#33928: [Reproducible Builds] Diff in generated/metadata between the builds

### CODE REVIEW

File: setup/src/Magento/Setup/Module/Di/Compiler/Config/Reader.php [hunk 1]
```diff
@@ -100,11 +100,6 @@ public function generateCachePerScope(
             $config['instanceTypes'][$virtualType] = $areaConfig->getInstanceType($virtualType);
         }
 
-        // sort configuration to have it in the same order on every build
-        ksort($config['arguments']);
-        ksort($config['preferences']);
-        ksort($config['instanceTypes']);
```
Comment (ihor-sviziev) [inline setup/src/Magento/Setup/Module/Di/Compiler/Config/Reader.php:106]: Note: this code was moved to `setup/src/Magento/Setup/Module/Di/App/Task/Operation/Area.php`

**Comment by BarnyShergold** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Note: we also have a diff in the `requirejs-config.js` file, but that's a bit another story

**Review by BarnyShergold — APPROVED** [review]
All looks good and the tests finally all passed! 😄

**Review by Den4ik — APPROVED** [review]

**Comment by sivaschenko** [comment]
Set P2 priority and moved to "Merge in progress" as this appears to be a good improvement for maintenance operations

**Pull request merged**